### PR TITLE
Issue 849 studioghibli

### LINKED ISSUE #849: Studio Ghibli API Integration
There is a really cool API with resources about Studio Ghibli's films. We could make a `s!ghibli` command!

https://ghibliapi.herokuapp.com/

### PR DESCRIPTION
**Pull request by merelyAnna**
Closes #849

### CODE REVIEW

File: src/commands/anime/studioghibli/vehicle.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      const filmData = await fetch(vehicleData.films[0]).then(res =>
+        res.json()
+      )
+
+      embed.addField('Name:', vehicleData.title, false)
+      embed.addField('Description:', vehicleData.description, false)
+      embed.addField('Vehicle Class:', vehicleData.vehicle_class, false)
+      embed.addField('Length:', vehicleData.length, false)
+      embed.addField('Pilot:', pilotData.name, false)
+      embed.addField('Films:', filmData.title, false)
```
Comment (pedrofracassi) [inline src/commands/anime/studioghibli/vehicle.js]: Field names should be made localizable using the `t` function. Example:
```js
embed.addField(t('commands:studioghibli.subcommands.vehicle.films'), filmData.title, false)
```

File: src/commands/anime/studioghibli/film.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    if (filmData === 'undefined') {
+      embed.setImage(t('commands:studioghibli.film.notFound'))
+    } else {
+      embed.addField('Name:', filmData.title, false)
+      embed.addField('Release Date:', filmData.release_date, false)
+      embed.addField('Director:', filmData.director, false)
+      embed.addField('Producter:', filmData.producter, false)
+      embed.addField('Rotten Tomatoes Score:', filmData.rt_score, false)
+      embed.addField('Description:', filmData.description, false)
```
Comment (pedrofracassi) [inline src/commands/anime/studioghibli/film.js]: Field names should be made localizable using the `t` function. Example:
```js
embed.addField(t('commands:studioghibli.subcommands.vehicle.films'), filmData.title, false)
```

File: src/commands/anime/studioghibli/characterlist.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+const { Command, SwitchbladeEmbed } = require('../../')
+
+const fetch = require('node-fetch')
+
+module.exports = class CharacterList extends Command {
+  constructor (client) {
+    super(client, {
+      name: 'characterList',
```
Comment (pedrofracassi) [inline src/commands/anime/studioghibli/characterlist.js]: No uppercase characters in command names.
```suggestion
      name: 'characterlist',
```

File: src/commands/anime/studioghibli/character.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    channel.startTyping()
+    const body = await fetch('https://ghibliapi.herokuapp.com/people').then(
+      res => res.json()
+    )
+    const characterData = body.filter(
+      element => element.name === searchTerm
+    )[0]
+
+    if (characterData === 'undefined') {
+      embed.setImage(t('commands:studioghibli.character.notFound'))
```
Comment (pedrofracassi) [inline src/commands/anime/studioghibli/character.js]: Why are you setting the image to text? That error text should be either the embed title or description, like so:
```js
embed.setDescription(t('commands:studioghibli.character.notFound'))
```

Actually, we have a thing called `CommandErrors`. They're throwable and require less code on the command. After including the class on the first line, you'd be able to use
```suggestion
      throw new CommandError(t('commands:studioghibli.subcommands.character.notFound'))
```

File: src/commands/anime/studioghibli/location.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+      embed.setImage(t('commands:studioghibli.location.notFound'))
+    } else {
+      const filmData = await fetch(locationData.films[0]).then(res =>
+        res.json()
+      )
+
+      embed.addField('Name:', locationData.name, false)
+      embed.addField('Climate:', locationData.climate, false)
+      embed.addField('Terrain:', locationData.terrain, false)
+      embed.addField('Film:', filmData.title, false)
```
Comment (pedrofracassi) [inline src/commands/anime/studioghibli/location.js]: Field names should be made localizable using the `t` function. Example:
```js
embed.addField(t('commands:studioghibli.subcommands.vehicle.films'), filmData.title, false)
```

File: src/commands/anime/studioghibli/species.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        const body = await fetch(film).then(res => res.json())
+        filmsList.push(body[0].title)
+      }
+
+      embed.addField('Name:', speciesData.name, false)
+      embed.addField('Classification:', speciesData.classification, false)
+      embed.addField('Eye Colors:', speciesData.eye_colors, false)
+      embed.addField('Hair Colors:', speciesData.hair_colors, false)
+      embed.addField('People:', peopleList.join(', '), false)
+      embed.addField('Films:', filmsList.join(', '), false)
```
Comment (pedrofracassi) [inline src/commands/anime/studioghibli/species.js]: Field names should be made localizable using the `t` function. Example:
```js
embed.addField(t('commands:studioghibli.subcommands.vehicle.films'), filmData.title, false)
```

File: src/commands/anime/studioghibli/character.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+      const filmData = await fetch(characterData.films[0]).then(res =>
+        res.json()
+      )
+
+      embed.addField('Name:', characterData.name, false)
+      embed.addField('Gender:', characterData.gender, false)
+      embed.addField('Eye Color:', characterData.eye_color, false)
+      embed.addField('Hair Color:', characterData.hair_color, false)
+      embed.addField('Species:', speciesData.name, false)
+      embed.addField('Film:', filmData.title, false)
```
Comment (pedrofracassi) [inline src/commands/anime/studioghibli/character.js]: Field names should be made localizable using the `t` function. Example:
```js
embed.addField(t('commands:studioghibli.subcommands.vehicle.films'), filmData.title, false)
```

**Comment by merelyAnna** [comment]
Thanks for your feedback. I'll edit the code when I'm back from walking my dog.

File: src/commands/anime/studioghibli/vehicle.js (hunk 1 above)
Comment (merelyAnna) [reply src/commands/anime/studioghibli/vehicle.js]: Added

File: src/commands/anime/studioghibli/film.js (hunk 2 above)
Comment (merelyAnna) [reply src/commands/anime/studioghibli/film.js]: Added

File: src/commands/anime/studioghibli/location.js (hunk 5 above)
Comment (merelyAnna) [reply src/commands/anime/studioghibli/location.js]: Added

File: src/commands/anime/studioghibli/species.js (hunk 6 above)
Comment (merelyAnna) [reply src/commands/anime/studioghibli/species.js]: Added

File: src/commands/anime/studioghibli/character.js (hunk 7 above)
Comment (merelyAnna) [reply src/commands/anime/studioghibli/character.js]: Added

**Comment by merelyAnna** [comment]
I would like to fix the failing check, but I am not sure why it is failing... Can any of you point me in the right direction?

**Comment by pedrofracassi** [comment]
Oh, that's the auto labeling action, it's not related to your code. I'll remove it from the repo

**Comment by pedrofracassi** [comment]
There we go. Can you pull the latest changes from `dev` to your branch, please?

**Comment by pedrofracassi** [comment]
Stale issue. Closing. Feel free to reopen it if you want.

**Pull request closed without merging**